### DISCUSSION TITLE
Thumbnails in BO lists are never regenerated

### DISCUSSION THREAD

**Discussion by zorrobiwan** [PrestaShop/PrestaShop#33419] [Ideas]
#### Describe the bug

When developing a module and displaying a simple list view of entries containing a picture on the fields_list, thumbnails are not regenerated. So, if you edit an entry, upload a new picture and save the entry, the thumbnail in the list will not be refreshed automatically.

#### Expected behavior

When displaying a list in the BO containing pictures, the thumbnails should be regenerated.

#### Steps to Reproduce

1. develop a module with an entity containing a pic and a thumbnail of this pic in the list view
2. edit an entity, upload a new pic, save the entity
3. see that the pic is not the good one :)

#### Additional information

* PrestaShop version: all versions
* PHP version: N/A

**Comment by zorrobiwan** [comment]
PR is solving the issue

**Comment by hibatallahAouadni** [comment]
Hello @zorrobiwan 

> develop a module with an entity containing a pic and a thumbnail of this pic in the list view

Could you please provide a test module to reproduce the issue :pray: 
Please check and feedback.

Thanks!

**Comment by zorrobiwan** [comment]
Hi there. Will do 😇

**Comment by khouloudbelguith** [comment]
Hi @zorrobiwan,

Thanks!
Waiting for your feedback.

Thank you!

**Comment by zorrobiwan** [comment]
[issue25716.zip](https://github.com/PrestaShop/PrestaShop/files/7066037/issue25716.zip)

Here is the module.

How to reproduce:
* install the module
* refresh the BO
* under Shop Parameters, choose "Issue 25716"
* Add a new item, enter a name (we don't care :) ), add a pic and Save
* Click on Modify, upload a new pic and save
=> the thumbnail is still from the previous pic

* Apply the patch
* edit the entry, upload a new pic, save
=> the thumbnail is regenerated and is now correct

**Comment by khouloudbelguith** [comment]
Hi @zorrobiwan,

Thanks for your feedback.
I think it is a module issue.

- I checked the carrier list => ok: https://drive.google.com/file/d/18iUnTSV6xyERoBqLCV7ZSQnJFRUl3Abn/view?usp=sharing
- I checked BO > Order Settings > Statuses list page  => ok https://drive.google.com/file/d/1ogEMfGU_D6MTl3rmoILgYqq5q1qfiYGe/view?usp=sharing
- I checked BO > Customer Settings > Titles list page => ok https://drive.google.com/file/d/1-l0Kr79TANVt_eKxQth9urAYe5k_pIMW/view?usp=sharing
- I checked the Customer reassurance module => ok https://drive.google.com/file/d/1HCeGCgHOx2Q7cM5IvdCWpEZ2XZ9tkXuB/view?usp=sharing

Thanks to check and feedback.

**Comment by zorrobiwan** [comment]
Thx @khouloudbelguith for your feedback and tests.

If you take a look to
https://github.com/PrestaShop/PrestaShop/blob/d9e2955167bc7f0115e3368d469ec3fb5f573418/controllers/admin/AdminGendersController.php#L217

You'll see the controller has to overide the afterImageUpload (coming from controller/AdminController and called when a image is uploaded) to delete the temporary thumbnail.
So, when ImageManager::thumbnail is called by classes/helper/HelperList displayListContent(), as the file does not exist anymore, it's regenerated.

_I don't understand why when coding an AdminController with a pic in the list view, you have to take care about deleting thumbnails on your own. For me, as the thumbnail creation is managed by the core classes, it's regeneration should be also managed by the core classes._

ImageManager::thumbnail proposes a way to force the regeneratation of the thumbnail but it's not used by the HelperList displayListContent() method (the two last paramaters are not specified - I guess the call has not been updated when paramters added to the method)

What do you think?

Thx :)

**Comment by khouloudbelguith** [comment]
@zorrobiwan, thanks for these clarifications.
I will ping our developers.
Ping @PrestaShop/prestashop-maintainers what do you think? is it a module issue or a core issue?

Thanks!

**Comment by kpodemski** [comment]
Hello @zorrobiwan 

At this moment, you need to delete the temporary image manually.

Your idea to have this done automatically is good, but the way you fix it is wrong. In my opinion, we should not force regeneration while visiting the list of items.

**Comment by zorrobiwan** [comment]
I agree it's not elegant way to do it. But doing it manually is not better.

In our case, lists are not so long, pic not so big so there is no perf issue.

Will think about a better way anyway

Thx for your feedback
